### DISCUSSION TITLE
Chain "..." does not support multicall

### DISCUSSION THREAD

**Discussion by ghost** [wevm/wagmi#935] [Q&A]
Hi everyone :)

I'm getting this warning on my console:

<img width="1710" alt="image" src="https://user-images.githubusercontent.com/109072720/189637116-197a1e6c-4673-431e-b91f-771118a1dcd2.png">

The warning comes from:

<img width="732" alt="image" src="https://user-images.githubusercontent.com/109072720/189634505-84eaa0d7-88e1-4d47-8bac-190fd8770027.png">

I'm pretty sure this comes from the "useContractReads" that I implement. Everything works well, and this doesn't make the app crash or anything like that, it's just something that pops up, non stop, in the console.

You can check the warning here:

https://app.audicity.io/

Thanks for the help :)

**Comment by tmm** [comment]
Did you include [`multicall`](https://github.com/wagmi-dev/wagmi/blob/main/packages/core/src/types/index.ts#L39-L42) in your `Chain` definition? You can see an [example here](https://github.com/wagmi-dev/wagmi/blob/main/packages/core/src/constants/chains.ts#L45).

Looks like the address for [multicall3](https://github.com/mds1/multicall) on Avalanche Fuji is [0xcA11bde05977b3631167028862bE2a173976CA11](https://testnet.snowtrace.io/address/0xcA11bde05977b3631167028862bE2a173976CA11#code).

**Reply by ghost** [reply]
> Did you include [`multicall`](https://github.com/wagmi-dev/wagmi/blob/main/packages/core/src/types/index.ts#L39-L42) in your `Chain` definition? You can see an [example here](https://github.com/wagmi-dev/wagmi/blob/main/packages/core/src/types/index.ts#L39-L42).
> 
> Looks like the address for [multicall3](https://github.com/wagmi-dev/wagmi/discussions/935) on Avalanche Fuji is [0xcA11bde05977b3631167028862bE2a173976CA11](https://testnet.snowtrace.io/address/0xcA11bde05977b3631167028862bE2a173976CA11#code).

Hi, thanks for the answer. Where did you find the info for the multicall address for Avalanche Fuji? I need to find the same for mainnet.

What happens with localhost, how does it affect?

I'm missing to understand exactly what does multicall mean/have to do. Is there any way it's nicely explained? I can imagine what it does, but to be sure.

And regarding the definition, i can see the "multicall" field is optional, but it has two params inside which seem to be obliged. What's the "blockCreated"?

Thanks for the help.

**Reply by tmm** [reply]
Edit: Linked [multicall3](https://github.com/mds1/multicall) above to correct location

- You can find multicall3 addresses here: https://github.com/mds1/multicall. Mainnet is built-in to wagmi and already has a multicall address ([see here](https://github.com/wagmi-dev/wagmi/blob/main/packages/core/src/constants/chains.ts#L45)).
- Localhost does not have a multicall address configured. If you use a tool, like [Anvil](https://github.com/foundry-rs/foundry/tree/master/anvil), you can fork a live network and provide the multicall address for that network to your localhost chain.
- > Multicall aggregates results from multiple contract constant function calls.
  > 
  > This reduces the number of separate JSON RPC requests that need to be sent (especially useful if using remote nodes like Infura), while also providing the guarantee that all values returned are from the same block (like an atomic read) and returning the block number the values are from (giving them important context so that results from old blocks can be ignored if they're from an out-of-date node).
- `blockCreated` is so wagmi knows when it is safe to use multicall or needs to fall back to firing off separate RPC requests
